### PR TITLE
Add tabindexes to toolbar elements and make sure a logical tab order is maintained

### PR DESCRIPTION
**Pull request by Stanton**
This fixes the following issues:

* Make sure toolbar elements are tabbed in the correct order
* Make sure main navigation is tabbed to once the user clicks the mobile menu button

Required to maintain compliance with WCAG 2.4.3 Focus Order: (Level A).

The following components will need a tabindex attribute adding.

| Component | Tabindex |
| ---- | ---- |
| `.skip-link` | 1 |
| `.jadu-branding` | 1 |
| `.nav-link` in primary nav | 1 |
| `.mobile-menu-button` | 2 |
| `.main-search__field` | 3 |
| `.main-search__submit` | 3 |
| `nav.help()` helper (user guides menu) | 4 |
| `.user-menu` | 5 |
| `.notifications-toggle` | 6 |

## Breaking change checklist

The following points should be considered as an early-warning of introducing a breaking change to a Continuum platform product. This is not an exhaustive list of what constitutes a breaking change.

| Does this pull request... | Yes / No |
| ---- | ---- |
| require changes to `main.js` | No |
| require changes to `index.js` | No |
| require changes to `pulsar.scss` | No |
| require changes to `gruntfile.js` | No |
| require changes to `package.json` (not including dev dependencies) | No |
| require changes to `base.html.twig` (or other core views like header/footer) | ⚠️Yes |
| require new arguments to be passed to a js component | No |
| require markup changes to maintain functionality | ⚠️Core elements listed above will need tabindex values adding to the base template |

## Browser testing checklist

- [x] Chrome
- [x] Safari - _requires option-tab or the 'press tab to highlight each item' option checked in preferences > advanced (this is default safari behaviour)_
- [x] Firefox
- [x] Edge
- [x] IE 11
- [x] IE 10
- [x] IE 9

### CODE REVIEW

**Comment by Stanton** [comment]
Would say this could go through as a minor, the changes to the base templates aren't breaking nor required to maintain functionality, they're just required to improve accessibility.

File: js/NavMainComponent.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ var $ = require('jquery');
 
 function NavMainComponent ($html, rootWindow) {
     this.$html = $html;
+    this.window = rootWindow;
     this.$window = $(rootWindow);
```
Comment (nvasileiadis) [inline js/NavMainComponent.js]: Could you please explain why we need both window and $window? What are the limitations of the code. Is it only for the stub in the testing? I don't understand that part

Comment (Stanton) [reply js/NavMainComponent.js]: `this.$window` already existed in this test but wouldn't work with matchMedia, which required the raw `window`. Looking at it now I can probably just pass `rootWindow` to matchMedia instead so I'll refactor 🤔

Comment (Stanton) [reply js/NavMainComponent.js]: not going to refactor actually, they're for different things, some functions use the jQuery object, matchMedia needs the raw window object

File: js/NavMainComponent.js [hunk 2]
```diff
@@ -46,9 +51,13 @@ NavMainComponent.prototype.init = function () {
         if ($self.text() === 'Menu') {
             $self.text('Close');
             $self.attr('aria-expanded', 'true');
+            component.$brandingLink.attr('tabindex', '3');
+            component.$primaryNavLinks.attr('tabindex', '3');
```
Comment (nvasileiadis) [inline js/NavMainComponent.js]: aXe finds them wrong. Elements should not have tabindex greater than zero -https://dequeuniversity.com/rules/axe/1.1/tabindex Is there another reason we use a greater number?

Comment (Stanton) [reply js/NavMainComponent.js]: aXe is rather strict for this condition, I believe this change to meet the 2.4.3 guideline.

From technique H4: Creating a logical tab order through links, form controls, and objects
> In some cases, the author may want to specify a tab order that follows relationships in the content without following the order of the interactive elements in the code. In these cases, an alternative order can be specified using the tabindex attribute of the interactive element. The tabindex is given a value between 0 and 32767.
https://www.w3.org/TR/WCAG20-TECHS/H4.html

And from F44: Failure of Success Criterion 2.4.3 due to using tabindex to create a tab order that does not preserve meaning and operability
> If tabindex is used, check that the tab order specified by the tabindex attributes follows relationships in the content.
https://www.w3.org/TR/WCAG20-TECHS/F44.html

Comment (Stanton) [reply js/NavMainComponent.js]: This has also been raised with aXe team: 
https://github.com/dequelabs/axe-core/issues/572

**Comment by Stanton** [comment]
Skip link added as first focusable item

**Pull request merged**